hibernate-core/src/main/java/org/hibernate/dialect/OracleDialect.java

```diff
@@ -704,82 +704,67 @@ public String getSelectGUIDString() {
-						// ORA-30006: resource busy; acquire with WAIT timeout expired
-						throw new LockTimeoutException(message, sqlException, sql);
-					case 54:
-						// ORA-00054: resource busy and acquire with NOWAIT specified or timeout expired
-						throw new LockTimeoutException(message, sqlException, sql);
-					case 4021:
-						// ORA-04021 timeout occurred while waiting to lock object
-						throw new LockTimeoutException(message, sqlException, sql);
-
-					// deadlocks ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-					case 60:
-						// ORA-00060: deadlock detected while waiting for resource
-						return new LockAcquisitionException( message, sqlException, sql );
-					case 4020:
-						// ORA-04020 deadlock detected while trying to lock object
-						return new LockAcquisitionException( message, sqlException, sql );
-
-					// query cancelled ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-					case 1013:
-						// ORA-01013: user requested cancel of current operation
-						throw new QueryTimeoutException(  message, sqlException, sql );
-
-					// data integrity violation ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-					case 1407:
-						// ORA-01407: cannot update column to NULL
-						final String constraintName = getViolatedConstraintNameExtracter().extractConstraintName( sqlException );
-						return new ConstraintViolationException( message, sqlException, sql, constraintName );
+		return (sqlException, message, sql) -> {
+			// interpreting Oracle exceptions is much much more precise based on their specific vendor codes.
+			switch ( JdbcExceptionHelper.extractErrorCode( sqlException ) ) {
 
-					default:
-						return null;
-				}
+				// lock timeouts ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+				case 30006:
+					// ORA-30006: resource busy; acquire with WAIT timeout expired
+					throw new LockTimeoutException(message, sqlException, sql);
+				case 54:
+					// ORA-00054: resource busy and acquire with NOWAIT specified or timeout expired
+					throw new LockTimeoutException(message, sqlException, sql);
+				case 4021:
+					// ORA-04021 timeout occurred while waiting to lock object
+					throw new LockTimeoutException(message, sqlException, sql);
+
+				// deadlocks ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+				case 60:
+					// ORA-00060: deadlock detected while waiting for resource
+					return new LockAcquisitionException( message, sqlException, sql );
+				case 4020:
+					// ORA-04020 deadlock detected while trying to lock object
+					return new LockAcquisitionException( message, sqlException, sql );
+
+				// query cancelled ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+				case 1013:
+					// ORA-01013: user requested cancel of current operation
```

Comment:
I beg pardon if I appears too picky, especially since English is not my first language.
So I learned the noun of 'cancel' and found this: https://www.wordhippo.com/what-is/the-noun-for/cancel.html
Seems 'cancellation' is better. 
Feel free to ignore the comment, however.

Reply:
If it were Javadoc I might care. But it's not worth a force push.